File: src/main/java/com/example/Toros/model/NewsItem.java
package com.example.Toros.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.Instant;

@Entity
@Table(name = "news")
@Data
public class NewsItem {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private String title;

    @Column(columnDefinition = "TEXT")
    private String summary;

    private String link;
    private String category; // e.g. "local", "major", "equipment", "app"
    private Instant publishedAt;
    private String source;

    /**
     * Optional URL for a cover image to display alongside the article.
     */
    private String coverImageUrl;
}

File: Toros.Backend/Toros.Backend/Models/NewsItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Toros.Backend.Models;

[Table("news")]
public class NewsItem
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public string Id { get; set; } = Guid.NewGuid().ToString();

    [Required]
    public string Title { get; set; } = string.Empty;

    [Column(TypeName = "text")]
    public string Summary { get; set; } = string.Empty;

    public string Link { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty; // e.g. "local", "major", "equipment", "app"
    public DateTime PublishedAt { get; set; }
    public string Source { get; set; } = string.Empty;

    public string? CoverImageUrl { get; set; }
}
